lib/extensions/sew_stack_editor.py

```diff
@@ -0,0 +1,545 @@
+
+        self.delete_layer_button = wx.Button(self.layer_list_wrapper, wx.ID_ANY, style=wx.BU_EXACTFIT)
+        self.delete_layer_button.SetBitmapLabel(wx.ArtProvider.GetBitmap(wx.ART_DELETE, wx.ART_MENU))
+        self.layer_buttons_sizer.Add(self.delete_layer_button, 0, wx.LEFT, 5)
+        self.delete_layer_button.Bind(wx.EVT_BUTTON, self.on_delete_layer_button)
+
+        self.move_layer_up_button = wx.Button(self.layer_list_wrapper, wx.ID_ANY, style=wx.BU_EXACTFIT)
+        self.move_layer_up_button.SetBitmapLabel(wx.ArtProvider.GetBitmap(wx.ART_GO_UP, wx.ART_MENU))
+        self.layer_buttons_sizer.Add(self.move_layer_up_button, 0, wx.LEFT, 5)
+        self.move_layer_up_button.Bind(wx.EVT_BUTTON, self.on_move_layer_up_button)
+
+        self.move_layer_down_button = wx.Button(self.layer_list_wrapper, wx.ID_ANY, style=wx.BU_EXACTFIT)
+        self.move_layer_down_button.SetBitmapLabel(wx.ArtProvider.GetBitmap(wx.ART_GO_DOWN, wx.ART_MENU))
+        self.layer_buttons_sizer.Add(self.move_layer_down_button, 0, wx.LEFT, 5)
+        self.move_layer_down_button.Bind(wx.EVT_BUTTON, self.on_move_layer_down_button)
+
+        self.layer_buttons_sizer.Add(0, 0, 1, wx.EXPAND)
+
+        self.single_layer_preview_button = wx.BitmapToggleButton(self.layer_list_wrapper, wx.ID_ANY, style=wx.BU_EXACTFIT | wx.BU_NOTEXT)
+        self.single_layer_preview_button.SetToolTip(_("Preview selected layer"))
+        self.single_layer_preview_button.SetBitmap(load_icon('layer', self))
+        self.single_layer_preview_button.Bind(wx.EVT_TOGGLEBUTTON, self.on_single_layer_preview_button)
+        self.layer_buttons_sizer.Add(self.single_layer_preview_button, 0, wx.LEFT, 0)
+
+        self.all_layers_preview_button = wx.BitmapToggleButton(self.layer_list_wrapper, wx.ID_ANY, style=wx.BU_EXACTFIT | wx.BU_NOTEXT)
+        self.all_layers_preview_button.SetToolTip(_("Preview all layers"))
+        self.all_layers_preview_button.SetBitmap(load_icon('layers', self))
+        self.all_layers_preview_button.SetValue(True)
+        self.all_layers_preview_button.Bind(wx.EVT_TOGGLEBUTTON, self.on_all_layers_preview_button)
+        self.layer_buttons_sizer.Add(self.all_layers_preview_button, 0, wx.LEFT, 1)
+
+        return self.layer_buttons_sizer
+
+    def update_layer_list(self):
+        self.layer_list.Freeze()
+
+        if self.layer_list.GetColumnCount() == 3:
+            # Save and restore column widths to work around an UltimateListCtrl bug.
+            # If we don't do this, ULC_AUTOSIZE_FILL stops working and the layer name
+            # column shrinks.
+            column_sizes = [self.layer_list.GetColumn(i).GetWidth() for i in range(self.layer_list.GetColumnCount())]
+        else:
+            column_sizes = (24, wx.LIST_AUTOSIZE, ulc.ULC_AUTOSIZE_FILL)
+
+        self.layer_list.ClearAll()
+
+        self.layer_list.InsertColumn(0, "", format=ulc.ULC_FORMAT_RIGHT)
+        self.layer_list.InsertColumn(1, _("Type"), format=ulc.ULC_FORMAT_CENTER)
```

Comment:
When I tested, I got a bunch of errors on OnPaint:
```
Traceback (most recent call last):
  File "/home/[redacted]/.local/lib/python3.10/site-packages/wx/lib/agw/ultimatelistctrl.py", line 5284, in OnPaint
    self.DrawTextFormatted(dc, text, wx.Rect(xAligned+EXTRA_WIDTH, HEADER_OFFSET_Y, cw-EXTRA_WIDTH, h-4))
TypeError: Rect(): arguments did not match any overloaded call:
  overload 1: too many arguments
  overload 2: argument 1 has unexpected type 'float'
  overload 3: argument 1 has unexpected type 'float'
  overload 4: argument 1 has unexpected type 'float'
  overload 5: argument 1 has unexpected type 'float'
  overload 6: argument 1 has unexpected type 'float'
```
This flag was the culprit, because if this is set then in `UltimateListHeaderWindow.OnPaint` it computes `xAligned` in a way that it can be a float:
```
            elif align == ULC_FORMAT_CENTER:
                xAligned = x + wcheck + (cw - wLabel)/2
```
This is a bug in ULC, really, but it still broke rendering for me. To get it to render right I had to change it to:
```
self.layer_list.InsertColumn(1, _("Type"))
```
Guessing you didn't get this error?

Reply:
Weird!  I didn't get this error.  I wonder if it's about your wxPython version?  I'm on 4.2.1.

Or maybe I just get lucky and mine's always been a float that could work as an int, and wx's argument type handler will cast a float to an int if it can?

I do want this column centered, I think...

Reply:
Though at some point I want to switch to showing the type as an icon, with every stitch layer type having an associated icon.  Maybe we would leave the header blank in that case and have the column be just one icon wide.

Reply:
Odd, I'm on wxPython 4.2.1 as well. To be precise what seems to have happened here is that on your system it renders that content as having an even width, and in my case it's odd. And because of python 3's division behaviour, an int divided by an int can be a float if it's not evenly divisible...

Imo remove the center flag for now and we can submit a fix to wxwidgets for the ULC center flag.

Reply:
Works for me!  I'm not at a computer right now but feel free to proceed with that.